app/controllers/concerns/inventory_service_helper.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Module to support controller integrations with the host inventory service
+module InventoryServiceHelper
+  extend ActiveSupport::Concern
+
+  included do
+    def add_inventory_hosts(ids)
+      existing_hosts = ::Pundit.policy_scope(current_user, ::Host)
+                               .where(id: ids)
+      save_hosts(ids - existing_hosts.pluck(:id)) + existing_hosts
```

Comment:
This part is slightly different - we only reach out to inventory if we need to. All hosts passed in are still returned from this method, including those that were already in our DB.